### PR TITLE
Add `mustache-curly-spacing` rule.

### PR DESCRIPTION
**Pull request by armano2**
This PR implements rule proposed in #150.

### CODE REVIEW

File: docs/rules/curly-bracket-spacing.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Enforce spacing on the style of curly brackets. (curly-bracket-spacing)
```
Comment (mysticatea) [inline docs/rules/curly-bracket-spacing.md]: I think that `curly-bracket` has too wide meaning. There are `object-curly-spacing`, `template-curly-spacing`, and `block-spacing` in core. How about `mustache-curly-spacing` or `embedded-expression-curly-spacing`?

File: docs/rules/curly-bracket-spacing.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+## :book: Rule Details
+
+This rule aims to enforce unified spacing of curly brackets.
+
+:-1: Examples of **incorrect** code for this rule:
+
+```html
+<template>
+  <div>{{   text   }}</div>
```
Comment (mysticatea) [inline docs/rules/curly-bracket-spacing.md]: It looks a correct code to me. It should be covered by `no-multi-spaces` rule.

File: tests/lib/rules/curly-bracket-spacing.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      filename: 'test.vue',
+      code: '<template><div>{{ text }}</div></template>',
+      options: ['always']
+    }
+  ],
+
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: '<template><div>{{         }}</div></template>',
```
Comment (mysticatea) [inline tests/lib/rules/curly-bracket-spacing.js]: I think that this rule should ignore mustaches which have syntax errors. If syntax errors exist, `node.expression` is `null`.

File: tests/lib/rules/curly-bracket-spacing.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    {
+      filename: 'test.vue',
+      code: '<template><div>{{text}}</div></template>',
+      options: ['never']
+    },
+    {
+      filename: 'test.vue',
+      code: '<template><div>{{ text }}</div></template>',
+      options: ['always']
+    }
```
Comment (mysticatea) [inline tests/lib/rules/curly-bracket-spacing.js]: Could you add tests which confirm that this rule doesn't warn directives, especially directives with unquoted attributes?

File: tests/lib/rules/curly-bracket-spacing.js (hunk 3 above)
Comment (armano2) [inline tests/lib/rules/curly-bracket-spacing.js]: ok

File: docs/rules/curly-bracket-spacing.md (hunk 1 above)
Comment (armano2) [inline docs/rules/curly-bracket-spacing.md]: i prefer `mustache-curly-spacing`.

File: tests/lib/rules/curly-bracket-spacing.js (hunk 4 above)
Comment (armano2) [inline tests/lib/rules/curly-bracket-spacing.js]: @mysticatea yup there was issue here, i forgot to check types of `startToken` and `endToken`

**Comment by armano2** [comment]
@mysticatea suggestion requested in code review applied

File: docs/rules/mustache-curly-spacing.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# Enforce spacing on the style of mustache curly brackets. (mustache-curly-spacing)
```
Comment (michalsnik) [inline docs/rules/mustache-curly-spacing.md]: Regarding rule name I have another proposition: `mustache-interpolation-spacing`. We all know what interpolation is, plus mustache interpolation type is well described in [official documentation](https://vuejs.org/v2/guide/syntax.html#Text) - this way we should avoid any confusion. What do you think @armano2 @mysticatea ?

File: docs/rules/mustache-curly-spacing.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+- :wrench: The `--fix` option on the [command line](http://eslint.org/docs/user-guide/command-line-interface#fix) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule aims to enforce unified spacing of curly brackets.
```
Comment (michalsnik) [inline docs/rules/mustache-curly-spacing.md]: `in mustache interpolations`

File: lib/rules/mustache-curly-spacing.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforce spacing on the style of mustache curly brackets.',
```
Comment (michalsnik) [inline lib/rules/mustache-curly-spacing.js]: `enforce unified spacing in mustache interpolations`

File: lib/rules/mustache-curly-spacing.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    // ----------------------------------------------------------------------
+
+    utils.registerTemplateBodyVisitor(context, {
+      'VExpressionContainer[expression!=null]' (node) {
+        const tokens = template.getTokens(node, {
+          includeComments: true,
+          filter: token => token.type !== 'HTMLWhitespace' // When there is only whitespace between ignore it
+        })
+
+        const startToken = tokens.shift()
```
Comment (michalsnik) [inline lib/rules/mustache-curly-spacing.js]: Do you intentionally mutate `tokens` here? Using that kind of methods usually introduce more confusion and may lead to unexpected results.

File: lib/rules/mustache-curly-spacing.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    const template = context.parserServices.getTemplateBodyTokenStore && context.parserServices.getTemplateBodyTokenStore()
+
+    // ----------------------------------------------------------------------
+    // Helpers
+    // ----------------------------------------------------------------------
+
+    function checkTokens (leftToken, rightToken) {
+      if (leftToken.loc.end.line === rightToken.loc.start.line) {
+        const spaces = rightToken.loc.start.column - leftToken.loc.end.column
+        if (optSpaces === (spaces === 0)) {
```
Comment (michalsnik) [inline lib/rules/mustache-curly-spacing.js]: Don't you think this expression is hard to read?
Can you please keep it simple stupid?
```
const noSpacesFound = spaces === 0
if (optSpaces && noSpacesFound) {
```

File: lib/rules/mustache-curly-spacing.js (hunk 8 above)
Comment (armano2) [inline lib/rules/mustache-curly-spacing.js]: Yes, i'm taking first and last token from array, checking their types, and i'm using it to compare

File: docs/rules/mustache-curly-spacing.md (hunk 5 above)
Comment (armano2) [inline docs/rules/mustache-curly-spacing.md]: I  not sure about the name of this rule, there was many suggestions like: curly-bracket, mustache-curly-spacing, embedded-expression-curly-spacing but i think yours matches at least documentation.

Comment (armano2) [inline docs/rules/mustache-curly-spacing.md]: @michalsnik I was not sure about the name of this rule, there was many suggestions like: curly-bracket, mustache-curly-spacing, embedded-expression-curly-spacing but i think yours matches at least documentation.

**Comment by armano2** [comment]
@michalsnik suggestions applied, thank you for code review.

**Pull request merged**